v2/contribute/decisions/oauth/0012-refresh-token-rotation.mdx

```diff
@@ -0,0 +1,42 @@
+---
+id: "0012"
+title: Refresh tokens are single-use only
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Refresh tokens are single-use only
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+Refresh token rotation is optional in OAuth2.0 but required in 2.1
+
+## Considered Options
+
+* No refresh token rotation
+* Always use rotating refresh tokens
+
+## Decision Outcome
+
+Chosen option: Always use rotating refresh tokens
+
+* Fits the newer security best practices
```

Comment:
Mention that we will not take care of many of the edge cases related to this and are OK with this.

Should we have a config that can be used to switch off refresh token rotation here? Maybe a core config? I can imagine that there might be users who just want to keep things simple and not care so much about security. Forcing them to also keep track of changing refresh tokens can be a big ask.

Reply:
Those users are most likely also using a lib that handles tokens for them, which should also support refresh tokens.

I'm not sure if this should be a core config or an oauth2 client level setting. The core config should be very easy to add later, so I think it's fine if we do not implement it at first.

Reply:
Sounds good. Please add this to the ADR, that we decided to purposely not add this core config for now and can do it later if required (as opposed to us just not having thought about it)